src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AbstractVoterTest.php

```diff
@@ -13,42 +13,59 @@
 
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Authorization\Voter\AbstractVoter;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
 
-/**
- * @author Roman Marintšenko <inoryy@gmail.com>
- */
 class AbstractVoterTest extends \PHPUnit_Framework_TestCase
 {
-    private $token;
+    protected $object;
+    protected $token;
 
     protected function setUp()
     {
-        $tokenMock = $this->getMock('Symfony\Component\Security\Core\Authentication\Token\TokenInterface');
-        $tokenMock
-            ->expects($this->any())
-            ->method('getUser')
-            ->will($this->returnValue('user'));
+        $this->object = $this->getMock('AbstractVoterTest_Object');
+        $this->token = $this->getMock('Symfony\Component\Security\Core\Authentication\Token\TokenInterface');
+    }
+
+    public function getTests()
+    {
+        $object = $this->getMock('AbstractVoterTest_Object');
```

Comment:
creating mocks in the data provider is a bad idea. These mocks will not be associated with the test properly (as they are created outside the test), which can lead to weird behaviors.